### ISSUE TITLE
nx 10.0.7: workspace-lint should respect .nxignore

### ISSUE THREAD

**Issue by tmtron** [nrwl/nx#3453]
## Current Behavior
`.nxignore` has no effect

## Expected Behavior
files listed in `.nxignore` should not cause errors when we run `nx workpace-lint`

## Steps to Reproduce
GitHub repository - branch `nxignore`: https://github.com/tmtron/nx-10-lint-issue/tree/nxignore
- clone the branch `nxignore`
- `npm install` 
- `nx workpace-lint`

we get this error:
```
>  NX   ERROR  The following file(s) do not belong to any projects:

  - libs/README.md

```

### Info
- the file [libs/README.md](https://github.com/tmtron/nx-10-lint-issue/blob/nxignore/libs/README.md) does not belong to any projct
- but it is mentioned in the [.nxignore](https://github.com/tmtron/nx-10-lint-issue/blob/nxignore/.nxignore) file and thus we should not get an error

## Misc
- This used to work in version 9.x.
- I've also added different path-variants - none worked:
```
libs/README.md
/libs/README.md
./libs/README.md
```
- Should nested `.nxignore` files work - or must we only have one at the project-root?
- Maybe this is related to #3376

### Environment
```
>  NX  Report complete - copy this into the issue template

  nx : Not Found
  @nrwl/angular : 10.0.7
  @nrwl/cli : 10.0.7
  @nrwl/cypress : 10.0.7
  @nrwl/eslint-plugin-nx : Not Found
  @nrwl/express : Not Found
  @nrwl/jest : 10.0.7
  @nrwl/linter : 10.0.7
  @nrwl/nest : 10.0.7
  @nrwl/next : Not Found
  @nrwl/node : 10.0.7
  @nrwl/react : Not Found
  @nrwl/schematics : Not Found
  @nrwl/tao : 10.0.7
  @nrwl/web : Not Found
  @nrwl/workspace : 10.0.7
  typescript : 3.9.7
```

**Comment by willyboy** [comment]
The root cause seems to be here: https://github.com/nrwl/nx/blob/master/packages/workspace/src/core/file-utils.ts#L221
When "usesGitForHashing" is true, nxignore is ignored. Tried putting the files in gitignore for kicks and giggles but that didn't work either.

**Comment by tmtron** [comment]
it seems that this was fixed in `10.0.13`

**Comment by savvyshell** [comment]
This is an issue again in 14.2.4
